python_ta/cfg/visitor.py

```diff
@@ -102,6 +102,42 @@ def visit_while(self, node: astroid.While) -> None:
         self.cfg.link_or_merge(end_else, after_while_block)
         self._current_block = after_while_block
 
+    def visit_for(self, node: astroid.For) -> None:
+        old_curr = self._current_block
+
+        # Handle "test" block
+        test_block = self.cfg.create_block()
+        test_block.add_statement(node.target)
+        test_block.add_statement(node.iter)
```

Comment:
Everything is good, except this should be merged with the block above it. The iterable is only evaluated once (before the loop), not at every iteration. You can keep the `node.target` where it currently is. Note that this will cause all the tests to be updated too.